.circleci/ci/ci.js

```diff
@@ -0,0 +1,167 @@
+     */
+    sh(command, returnStdout = false, print = true) {
+        if (print) {
+            console.log(command);
+        }
+        if (returnStdout) {
+            return e.execSync(command).toString().trim();
+        }
+        return e.execSync(command, {stdio: 'inherit'});
+    };
+
+    /**
+     * Return value of given environment variable.
+     */
+    env(key) {
+        return process.env[key];
+    }
+
+    /**
+     * Print stage name.
+     */
+    stage(name) {
+        console.log("\n------------------------------\n" +
+            "--\n" +
+            "-- %s\n" +
+            "--\n" +
+            "------------------------------\n", name);
+    };
+
+    /**
+     * Configure a git impersonation for the scope of the given function.
+     */
+    gitImpersonate(user, mail, func) {
+        try {
+            this.sh('git config --local user.name ' + user + ' && git config --local user.email ' + mail, false, false);
+            func()
+        } finally {
+            this.sh('git config --local --unset user.name && git config --local --unset user.email', false, false);
+        }
+    };
+
+    /**
+     * Configure git credentials for the scope of the given function.
+     */
+    gitCredentials(repo, func) {
+        try {
+            this.sh('git config credential.helper \'store --file .git-credentials\'');
+            fs.writeFileSync('.git-credentials', repo);
+            console.log('// Created file .git-credentials.');
+            func()
+        } finally {
+            this.sh('git config --unset credential.helper');
+            fs.unlinkSync('.git-credentials');
+            console.log('// Deleted file .git-credentials.');
+        }
+    };
+
+    /**
+     * Writes given content to a file.
+     */
+    writeFile(fileName, content) {
+        console.log(`// Write to file ${fileName}`);
+        fs.writeFileSync(fileName, content, { 'encoding': 'utf8' });
+    }
+
+    collectConfiguration() {
+        let configuration = {
+            modules: {}
+        };
+        
+        let folders = [process.cwd()];
+        process.stdout.write("Collecting project configuration");
+        while(folders.length) {
+            let folder = folders.shift();
+            let files = fs.readdirSync(folder, { withFileTypes: true });
+        
+            for(let file of files) {
+                if (file.isDirectory()) {
+                    folders.push(path.resolve(folder, file.name));
+                    continue;
+                }
+        
+                if (file.name !== 'pom.xml') {
+                    continue;
+                }
+        
+                let pomPath = path.resolve(folder, file.name);
+                let metaData = this.sh('printf \'${project.groupId}|${project.artifactId}|${project.name}|${project.version}|${project.packaging}\' | mvn -f ' + pomPath + ' help:evaluate --non-recursive | grep -Ev "(Download|\\[)"', true, false).split('|');
```

Comment:
You could also maybe use https://www.npmjs.com/package/pom-parser

Reply:
For now I copied the command from the Sprouts, but you're right. Since we're using a node script to execute commands, we can do further optimisations for pom parsing. Let's integrate those step by step :)